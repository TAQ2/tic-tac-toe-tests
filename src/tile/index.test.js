import React from "react";
import { shallow } from "enzyme";
import "jest-styled-components";

import Tile from "../tile";

describe("Tile component", () => {
  it("displays a token", () => {
    const token = "bob";
    const wrapper = shallow(<Tile token={token} />);

    expect(wrapper.contains(token)).toEqual(true);
  });

  it("Tile has an onClick", () => {
    const mockFunction = jest.fn();
    const wrapper = shallow(<Tile onTileClick={mockFunction} />);

    wrapper.simulate("click");

    expect(mockFunction).toBeCalled();

    expect(wrapper).toMatchSnapshot();
  });
});
