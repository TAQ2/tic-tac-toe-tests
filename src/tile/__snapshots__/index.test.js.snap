// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tile component Tile has an onClick 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Tile
    onTileClick={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <styled.rect
          height={NaN}
          width={NaN}
          x={NaN}
          y={NaN}
/>,
        <styled.text
          x={NaN}
          y={NaN}
/>,
      ],
      "className": "tileundefinedundefined",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "height": NaN,
          "width": NaN,
          "x": NaN,
          "y": NaN,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": undefined,
          "x": NaN,
          "y": NaN,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "g",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <styled.rect
            height={NaN}
            width={NaN}
            x={NaN}
            y={NaN}
/>,
          <styled.text
            x={NaN}
            y={NaN}
/>,
        ],
        "className": "tileundefinedundefined",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "height": NaN,
            "width": NaN,
            "x": NaN,
            "y": NaN,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": undefined,
            "x": NaN,
            "y": NaN,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "g",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
